ddtrace/internal/writer/writer.py

```diff
@@ -385,27 +385,35 @@ def flush_queue(self, raise_exc: bool = False):
     def _flush_queue_with_client(self, client: WriterClientBase, raise_exc: bool = False) -> None:
         n_traces = len(client.encoder)
         try:
-            encoded, n_traces = client.encoder.encode()
-
-            if encoded is None:
+            if not (encoded_traces := client.encoder.encode()):
                 return
 
-            # Should gzip the payload if intake accepts it
-            if self._intake_accepts_gzip:
-                original_size = len(encoded)
-                # Replace the value to send with the gzipped the value
-                encoded = gzip.compress(encoded, compresslevel=6)
```

Comment:
We are moving these operation out of a Try-Except. Are there cases where len(encoded) or gzip.compress can raise an exception (ex: encoded is None)? An exception raised by these operation could cause the writer crash.

Reply:
Good point!

Reply:
Fixed, and added a regression test, thank you!